clientconn.go

```diff
@@ -412,6 +413,14 @@ func WithChannelzParentID(id int64) DialOption {
 	}
 }
 
+// WithServiceConfigOff returns a DialOption that specifies whether resolver
+// should fetch service config.
+func WithServiceConfigOff(a bool) DialOption {
```

Comment:
What if we made this `WithResolverBuilderOptions(opts resolver.BuilderOption)` instead, to be more extensible?

@menghanl

Reply:
But what if we later want to add another field to `resolver.BuilderOption` that's not to be configured by the user?